src/azul/lambdas.py

```diff
@@ -97,21 +100,42 @@ class Lambdas:
     def _lambda(self):
         return aws.lambda_
 
-    def list_lambdas(self) -> list[Lambda]:
+    def list_lambdas(self,
+                     deployment: str | None = None,
+                     all_versions: bool = False
+                     ) -> list[Lambda]:
+        """
+        Return a list of AWS Lambda functions
+
+        :param deployment: Limit output to the specified deployment stage. If
+                           `None`, functions from all deployments will be
+                           returned.
+
+        :param all_versions: If `True`, return all versions of each AWS Lambda
+                             function (including '$LATEST'). If `False`, return
+                             only the latest version of each function.
+        """
+        paginator = self._lambda.get_paginator('list_functions')
+        lambda_prefixes = [
+            config.qualified_resource_name(lambda_infix, stage=deployment)
+            for lambda_infix in config.lambda_names()
+            if deployment is not None
+        ]
+        assert all(lambda_prefixes)
+        params = {'FunctionVersion': 'ALL'} if all_versions else {}
         return [
             Lambda.from_response(function)
-            for response in self._lambda.get_paginator('list_functions').paginate()
+            for response in paginator.paginate(**params)
             for function in response['Functions']
+            if deployment is None or any(
+                function['FunctionName'].startswith(prefix)
+                for prefix in lambda_prefixes
+            )
         ]
 
     def manage_lambdas(self, enabled: bool):
-        paginator = self._lambda.get_paginator('list_functions')
-        lambda_prefixes = [config.qualified_resource_name(lambda_infix) for lambda_infix in config.lambda_names()]
-        assert all(lambda_prefixes)
-        for lambda_page in paginator.paginate(FunctionVersion='ALL', MaxItems=500):
-            for lambda_name in [metadata['FunctionName'] for metadata in lambda_page['Functions']]:
-                if any(lambda_name.startswith(prefix) for prefix in lambda_prefixes):
-                    self.manage_lambda(lambda_name, enabled)
+        for lambda_ in self.list_lambdas(deployment=config.deployment_stage):
```

Comment:
```suggestion
        for function in self.list_lambdas(deployment=config.deployment_stage):
```

As noted before, we are phasing this term out. You don't have to remove all usages but just don't add new ones.